toolkit.py

```diff
@@ -0,0 +1,142 @@
+                           dtype=cv.CV_32S)
+        y = np.diff(map(lambda *row: list(row), *vlines)[0])
+        x = np.diff(hlines[0])
+        return (self.__get_index(x), self.__get_index(y))
+
+    def get_table_metadata(self, image):
+        """
+        Function for returning coordinates of
+        a boards corners. It distorts image with
+        median blur to remove noise from image and
+        unify colors a bit.
+        """
+        process_image = cv.medianBlur(image, 5)
+        return self.calc_hist(process_image)
+
+    def write_video_metadata(self, video, fd):
+        """
+        Sets up a video capture and loops through
+        all of the frames. Pulls frame metadata
+        from every frame and stores it into an array.
+        After we are done with a video we flush these
+        values to a file with frame index.
+        """
+        vid = cv.VideoCapture(video)
+        if vid is None:
+            print "Couldn't load video {}".format(video)
+            return
+        frames = []
+        while(vid.isOpened()):
+            ret, frame = vid.read()
+            if not ret:
+                break
+            (x, y) = self.get_table_metadata(frame)
+            frames.append((x, y))
+        for i, frame in enumerate(frames):
+            fd.write("{}: {} {}\n".format(i, frame[0], frame[1]))
+        vid.release()
+
+    def __get_file_name(self, path):
+        """
+        Returns only a file name from path.
+        """
+        head, tail = ntpath.split(path)
+        name = tail or ntpath.basename(head)
+        return name.split('.')[0]
+
+    def write_image_metadata(self, image, fd):
+        """
+        Reads given image and gets his metadata which
+        are written to a file.
+        """
+        im = cv.imread(image, cv.CV_LOAD_IMAGE_COLOR)
+        if im is None:
+            print "Couldn't read image file {}".format(image)
+            return
+        (x, y) = self.get_table_metadata(im)
+        fd.write("0: {} {}".format(x, y))
+
+    image_extension_list = ["jpg", "gif", "png"]
+    video_extension_list = ["mkv", "wmv", "avi", "mp4"]
+
+    @timeit
+    def write_metadata(self, input_file):
+        """
+        Main function for getting metadata from images or videos.
+        It checks if format of a file is supported and then calls
+        approperiate function.
+        """
+        fd = open("{}_matadata".format(self.__get_file_name(input_file)), "w+")
+        if(any(input_file[-3:] == i for i in self.video_extension_list)):
+            self.write_video_metadata(input_file, fd)
+        elif (any(input_file[-3:] == i for i in self.image_extension_list)):
+            self.write_image_metadata(input_file, fd)
+        else:
+            print "Unrecognized file format"
+        fd.close()
+
+
+def main(input_file):
+    toolkit = BoardToolkit()
+    toolkit.write_metadata(input_file)
+
+if __name__ == '__main__':
+    main(sys.argv[1])
```

Comment:
@Foxiy this is nasty. Even though it is just a helper, it would be nice to have some sort of usage, so that when one runs it she does not have to check the code to figure out what should the argument be (and whether there should be any).